src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ProjectConfigurationFactory.cs

```diff
@@ -28,9 +29,9 @@ public static ProjectConfiguration Create(string configuration)
             return Create(configuration, dimensionsBuilder.ToImmutable());
         }
 
-        public static IImmutableSet<ProjectConfiguration> CreateMany(params string[] configurations)
+        public static IReadOnlyList<ProjectConfiguration> CreateMany(params string[] configurations)
         {
-            var configurationsBuilder = ImmutableHashSet.CreateBuilder<ProjectConfiguration>();
+            var configurationsBuilder = ImmutableArray.CreateBuilder<ProjectConfiguration>();
```

Comment:
In theory the hash set would have enforced that the same `ProjectConfiguration` wasn't added twice, assuming the underlying implementation of the `ProjectConfiguration` interface actually implemented a proper `GetHashCode`.

Does this need to be enforced in some other way within `CreateMany` or do we just say that the caller needs to ensure there are no duplicates in the string array it passes in?